chronos/asyncloop.nim

```diff
@@ -1100,6 +1102,15 @@ proc wait*[T](fut: Future[T], timeout = InfiniteDuration): Future[T] =
 
   return retFuture
 
+proc wait*[T, E](fut: RaiseTrackingFuture[T, E], timeout = InfiniteDuration): auto =
```

Comment:
what's the return type here? in public functions, we ideally want to avoid auto both for documentation reasons and because it serves as a type inference guide for the compiler.

Reply:
It's RTF[T, E + CanceledError] (which I just realized is wrong, should be AsyncTimeoutError)

I don't know a better solution than auto for this kind of procedures